### ISSUE TITLE
[java] LocalVariableCouldBeFinal faslse positive on var assigned in try and catch

### ISSUE THREAD

**Issue by zman0900** [pmd/pmd#5060]
**Affects PMD Version:** 7.2.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#localvariablecouldbefinal>

**Description:**
PMD 7.2.0 reports that a local variable could be final, but doing so causes compile failure with "variable might already have been assigned" error.

**Code Sample demonstrating the issue:**

```java
public int toInt(final String value) {
    int result; // <-- reported as could be final, but adding 'final' does not compile
    try {
        result = Integer.parseInt(value);
    } catch (NumberFormatException e) {
        result = 0;
    }
}
```

**Expected outcome:**

Expected no violation reported for LocalVariableCouldBeFinal rule.

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
Duplicate of #5046
